File: LeetCode_2044/Main.c
#include<stdio.h>

int countMaxOrSubsets(int* nums, int n) {
    int maxOr = 0, cnt = 0;
    int sz = 1 << n;
    
    for (int mask = 0; mask < sz; ++mask) {
        int cur = 0;
        for (int i = 0; i < n; ++i) {
            if (mask & (1 << i)) {
                cur |= nums[i];
            }
        }
        if (cur > maxOr) {
            maxOr = cur;
            cnt = 1;
        } else if (cur == maxOr) {
            cnt++;
        }
    }
    return cnt;
}

File: LeetCode_2044/Main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countMaxOrSubsets = function(nums) {
    let maxOr = 0, count = 0;
    let n = nums.length;
    let total = 1 << n;

    for (let mask = 0; mask < total; mask++) {
        let cur = 0;
        for (let i = 0; i < n; i++) {
            if ((mask & (1 << i)) !== 0) {
                cur |= nums[i];
            }
        }
        if (cur > maxOr) {
            maxOr = cur;
            count = 1;
        } else if (cur === maxOr) {
            count++;
        }
    }
    return count;
};
